### PR TITLE
Add back button to Reloadable screen

### PR DESCRIPTION
**Pull request by akalin-keybase**
This is needed on mobile for anything in settings, as otherwise the user
wouldn't be able to go back to the main settings page.

### CODE REVIEW

**Comment by akalin-keybase** [comment]
cc @MarcoPolo who's adding Reloadable to other things in Settings.

File: shared/common-adapters/reload.js [hunk 1]
```diff
@@ -100,6 +103,14 @@ const styles = Styles.styleSheetCreate({
   },
 })
 
+type OwnProps = {|
+  children: React.Node,
+  onBack?: () => void,
```
Comment (MarcoPolo) [inline shared/common-adapters/reload.js]: Should this be optional? If it has the potential to get you in a broken state on mobile maybe it should be required.

Comment (akalin-keybase) [reply shared/common-adapters/reload.js]: it's optional because it doesn't apply on desktop, right?

**Comment by akalin-keybase** [comment]
I did it as a prop because I thought global `navigateUp` was racy, so it's safer to pass it down unless you have to pass it down really deep.

File: shared/common-adapters/reload.js (hunk 1 above)
Comment (MarcoPolo) [reply shared/common-adapters/reload.js]: Yeah, but it wouldn't hurt to have it on desktop. I imagine it'll be more common for this component to live in the shared code than not, so we should make it so the types will make the component work properly in both.

**Pull request merged**